### DISCUSSION TITLE
[solved] Plugin External Links - exclude btn class

### DISCUSSION THREAD

**Discussion by raramuridesign** [GetPublii/Publii#1533] [Ideas]
### Operating system

Windows

### Publii version

Version: 0.45.2 (build 16609)

### Issue type

Free plugin

### Bug description

https://marketplace.getpublii.com/plugins/external-links/

Hi Team @dziudek 
Thank you for creating the new plugin, works well.

This is more of a suggestion than a bug. Could you add the ability to exclude or include the icon on buttons.

See screenshot. 
![image](https://github.com/GetPublii/Publii/assets/902376/47561e06-c4ec-4ce4-9ebc-fc724811f419)

Would it be possible to add this.
Thanks
M

**Comment by bobmitro** [comment]
I'm glad to hear that the plugin is working well for you. Regarding your suggestion, unfortunately, it's not possible to add the ability to exclude or include icons on buttons. This is because buttons are not links, and the plugin is specifically designed to enhance external links only. It focuses on adding icons, rel attributes, additional CSS classes, and titles to external links, which have distinct attributes such as href or rel that buttons lack.

PS: you can easily add the icon to your buttons with the Feather Icons plugin:  https://marketplace.getpublii.com/plugins/feather-icons/

**Comment by raramuridesign** [comment]
@bobmitro  Thanks for the info.
The <button> element makes sense but we are using a class on the a element which generates a button look & feel, as per the typography of the templates. The code looks like this.

```<p><a href="https://domain.com/" target="_blank" rel="noopener noreferrer" class="btn">External Link</a></p>```

The `btn` class shows a button style. and this is where it would be good to exclude this.

Does this make sense
M.

**Comment by bobmitro** [comment]
So, this is a link, not a button. Select the "No Icon" option to ensure the icon does not appear in your link.

**Reply by raramuridesign** [reply]
Yes, that I understand, but what about normal links on the page - see my original screenshot - where I want this to appear ;-)

**Reply by bobmitro** [reply]
I have no idea :)

**Reply by bobmitro** [reply]
I've just added the feature you requested. You can now exclude specific CSS classes and IDs from being modified by the plugin, allowing you greater control over which links are affected. 

Please download version 1.1.0 to access this new feature. Here's the download link: [External Links Plugin v1.1.0](https://marketplace.getpublii.com/plugins/external-links/)

**Reply by raramuridesign** [reply]
Thanks @bobmitro Works like a charm, appreciate the fast turn around on this. 
Adding the id as an exclusion is also a good idea.
